src/Iris/Cli/Interactive.hs

```diff
@@ -46,3 +50,21 @@ interactiveModeP = nonInteractiveP <|> pure Interactive
         [ Opt.long "no-input"
         , Opt.help "Enter the terminal in non-interactive mode"
         ]
+
+{- | Forces non interactive mode when the terminal is not interactive
+
+Use this function to check whether you can get input from the terminal:
+
+@
+'handleInteractiveMode' 'requestedInteractiveMode'
+@
+
+If the terminal is non interactive i.e. the program is run in a pipe, interactive mode is set to false no matter what
+
+@since 0.0.0.1
+-}
+handleInteractiveMode :: InteractiveMode -> IO InteractiveMode
+handleInteractiveMode optionMode = do
+    isTerminal <- hIsTerminalDevice stdin -- misleading on windows where some terminals are emulated
+    supportsANSI <-  hSupportsANSI stdin -- false on dumb terminal, better windows support
+    pure $ if supportsANSI || isTerminal  then optionMode else NonInteractive
```

Comment:
[According to the implementation](https://hackage.haskell.org/package/ansi-terminal-0.11.3/docs/src/System.Console.ANSI.Unix.html#hSupportsANSI), `hSupportsANSI` is implemented as `hIsTerminalDevice && isNotDumb`. So this check becomes:

```haskell
hIsTerminalDevice || (hIsTerminalDevice && isNotDumb)
```

which can be simplified to just:

```haskell
hIsTerminalDevice
```

So the `hSupportsANSI` check is redundant

Reply:
except has a completely different implementation in windows:

```haskell
-- hSupportsANSIWithoutEmulation :: Handle -> IO (Maybe Bool)
-- (See Common-Include.hs for Haddock documentation)
hSupportsANSIWithoutEmulation = E.hSupportsANSIWithoutEmulation


-- hSupportsANSIWithoutEmulation :: Handle -> IO (Maybe Bool)
-- (See Common-Include.hs for Haddock documentation)
hSupportsANSIWithoutEmulation handle = do
  supportsANSI <- detectHandleSupportsANSI handle  -- Without reference to the
                                                   -- environment
  case supportsANSI of
    Just isSupported -> return (Just isSupported)
    Nothing -> do  -- Not sure, based on the handle alone
      notDumb <- isNotDumb  -- Test the environment
      if notDumb
        then return Nothing  -- Still not sure!
        else return (Just False) -- A dumb terminal
```